### PR TITLE
Common Hook Set Up Steps Guide - to one page

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Updated Common Hook Set Up Steps guide to one guide 
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-438763](https://oktainc.atlassian.net/browse/OKTA-438763)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+---
+
+**Learning outcomes**
+
+* Understand common set up steps for implementing an Okta Event or Inline Hook
+* Use these steps when running the example hook code in the accompanying guides
+
+**What you need**
+
+* A [Glitch.com](https://glitch.com) project or account.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:18]:
```suggestion
* A [Glitch.com](https://glitch.com) project or account
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* An Okta developer org. [Create an org for free](https://developer.okta.com/signup/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:19]:
```suggestion
* An Okta developer org. [Create an org for free](https://developer.okta.com/signup/)
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+---
+
+This guide explains common set-up steps when implementing an Okta Event or Inline hook, including using Glitch.com as an example external service, adding basic authentication to the hook calls, JSON body parsing in the external service code, and troubleshooting steps.
+
+---
+
+**Learning outcomes**
+
+* Understand common set up steps for implementing an Okta Event or Inline Hook
+* Use these steps when running the example hook code in the accompanying guides
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:14]:
```suggestion
* Understand how to use these steps when running the example hook code in the accompanying guides
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+[Glitch](https://www.glitch.com) is a browser-based development environment that can build a full-stack web application online. You can use their template applications to implement an external service that receives the outbound calls from Okta orgs.
+
+Start with a new Node.js project built on the Express framework or a Node.js SQLite database application and use the code snippets in the following examples to implement the example hooks. Copy (**Remix on Glitch**) the Glitch projects for each hook in the following sections to have a working code sample.
+
+<StackSelector snippet="setup"/>
+
+## Add Basic Authorization and Body Parsing
+
+The Glitch project templates do not have any authorization or body parsing code. To include this content:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:49]:
```suggestion
The Glitch project templates don't have any authorization or body parsing code. To include this content:
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+* Add the Body Parser and Basic Auth `npm` packages to your Glitch project
+* Add the code snippet below
+
+If you Remix a Glitch Inline Hook project, the packages and code are already included.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:54]: Should Remix a Glitch Inline Hook project have caps? If so, should it be in quotes or something? It reads kind of funny this way...
```suggestion
If you Remix a Glitch Inline Hook project, the packages and code are already included.
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+To add the `npm` packages:
+
+1. Select the `package.json` file in the left-hand project menu.
+2. From the `Add Package` drop-down, search for the `express-basic-auth` and `body-parser` packages.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:59]:
```suggestion
2. From the `Add Package` drop-down box, search for the `express-basic-auth` and `body-parser` packages.
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+3. Click each package to add to your project.
+
+The Inline Hook guides use [HTTP Basic Authentication](/books/api-security/authn/api-authentication-options/#http-basic-authentication) to authenticate the Okta Inline Hook API calls received by your Glitch external service. In your Okta org, the Glitch project username and password credentials must be encoded in Base64 and added as the **Authentication secret** when you activate the Inline Hook. Ensure you add the scheme `Basic ` (including a space) as a prefix to the **Authentication secret** value.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:62]:
```suggestion
The Inline Hook guides use [HTTP Basic Authentication](/books/api-security/authn/api-authentication-options/#http-basic-authentication) to authenticate the Okta Inline Hook API calls received by your Glitch external service. In your Okta org, you must encode the Glitch project username and password credentials in Base64 and add them as the **Authentication secret** when you activate the Inline Hook. Ensure that you add the scheme `Basic ` (including a space) as a prefix to the **Authentication secret** value.
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+<StackSelector snippet="auth"/>
+
+## Troubleshoot hook implementations
+
+After setting up an external service and an Event Hook or Inline Hook, you may need to troubleshoot or review your configurations. Use the following options to confirm a successful implementation.
+
+### Preview tab
+
+An [Inline Hook Preview](https://help.okta.com/okta_help.htm?id=ext-preview-inline-hooks)tab, accessible in the Admin Console, is available for the following two Inline Hooks:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:76]:
```suggestion
An [Inline Hook Preview](https://help.okta.com/okta_help.htm?id=ext-preview-inline-hooks) tab, accessible in the Admin Console, is available for the following two Inline Hooks:
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+* Registration Inline Hook
+* SAML Inline  Hook
+
+Before enabling the hook, the preview tab can run a sample Okta request call, and receive the external service response. Review the request and response formats to make sure responses are accurate.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:81]:
```suggestion
Before enabling the hook, the preview tab can run a sample Okta request call and receive the external service response. Review the request and response formats to make sure responses are accurate.
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+An [Event Hook Preview](https://help.okta.com/okta_help.htm?id=ext-event-hooks-preview) tab is also available for Event Hooks, and displays the JSON payload for the selected Event Type. The preview tab can confirm a successful delivery of the request.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:83]:
```suggestion
An [Event Hook Preview](https://help.okta.com/okta_help.htm?id=ext-event-hooks-preview) tab is also available for Event Hooks and displays the JSON payload for the selected Event Type. The preview tab can confirm a successful delivery of the request.
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+### Admin Console System Log
+Use the Admin Console System Log to review logs of the Event, Inline Hook triggers, or errors encountered during testing from the Okta org. See [System Log](https://help.okta.com/okta_help.htm?id=ext_Reports_SysLog)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:86]:
```suggestion
Use the Admin Console System Log to review logs of the Event, Inline Hook triggers, or errors encountered during testing from the Okta org. See [System Log](https://help.okta.com/okta_help.htm?id=ext_Reports_SysLog).
```

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+For implementations using the Glitch projects, use Glitch's log feature to review and troubleshoot your external service code:
+
+1. In the Glitch project's left-hand folder navigation pane, click **Tools** at the bottom of the pane.
+2. Click **Logs**.
+
+A log pane appears that displays all `console.log()` output. Some console output code is available in the sample code.
+
+## Next steps
+
+Review the following guides to implement a simple hook example, and preview the hook functionality:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:99]:
```suggestion
Review the following guides to implement a simple hook example and preview the hook functionality:
```

File: packages/@okta/vuepress-site/docs/guides/event-hook-implementation/initial-verification/index.md [hunk 13]
```diff
@@ -5,7 +5,7 @@ Okta Event Hooks require an initial verification of the external service endpoin
 
 Add the following code to your external service to address this request.
 
-> **Note:** Also, make sure to have the required default code and packages in your project. See [Overview and considerations](/docs/guides/common-hook-set-up-steps/overview) for further information.
+> **Note:** Also, make sure to have the required default code and packages in your project. See [Overview and considerations](/docs/guides/common-hook-set-up-steps/) for further information.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/event-hook-implementation/initial-verification/index.md:8]:
```suggestion
> **Note:** Also, make sure to have the required default code and packages in your project. See [Overview and considerations](/docs/guides/common-hook-set-up-steps/).
```

**Review by susanharper-okta — COMMENTED** [review]
couple things

File: packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md (hunk 5 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/main/index.md:54]: Good point, I believe I was trying to use the Glitch UI terminology. I'll lower case the "Remix" ... everything else should stay capitalized.

**Comment by brianduffield-okta** [comment]
Thanks for catching all these, Susan.

**Review by susanharper-okta — APPROVED** [review]
LGTM 🚢

**Pull request merged**